src/gui/qgsbrowserdockwidget.h

```diff
@@ -102,6 +102,8 @@ class GUI_EXPORT QgsBrowserDockWidget : public QgsDockWidget, private Ui::QgsBro
     void openFile( const QString & );
     //! Emitted when drop uri list needs to be handled
     void handleDropUriList( const QgsMimeDataUtils::UriList & );
+    //! Connections changed in the browser
+    void connectionsChanged( );
```

Comment:
@wonder-sk thanks for the comments, I'll try to follow your suggestions, indeed I plan to move more common functionalities stuff into the base class as long as I move to refactor the other provider dialogs.

I'm not sure about dropping completely the possibility to use the dialogs as standalone, don't you think that they still could be useful in some particular situations where the type of layer is know (or constrained) in advance?

I'm blocked on the arcgis stuff though: I've already wasted quite some trying to move  the arcgis dialog to the provider as suggested by @manisandro but the canvas dependency stepped on my toes again: that dependency implies that the class must be available and included in the data source manager dialog code, that is in now in GUI, and I couldn't find the right way (well, *a* way) to include a provider's class (the moved arcgis dialog) from GUI. Maybe @manisandro (sorry Sandro to bother you again about this canvas dependency) knows a way to remove the dependency?

As a possible workaround, for example I was thinking of an additional (empty by default) virtual method in the base class, the sets the extent and CRS, that method would be called from the app (or whatever) to set the extent and CRS . This would also remove the canvas argument from the data source manager dialog, that is really only used for the arcgis provider.